packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-dynamo-model-resource-generator.ts

```diff
@@ -58,52 +68,82 @@ export class AmplifyDynamoModelResourceGenerator extends DynamoModelResourceGene
             'dynamodb:UpdateTimeToLive',
           ],
           resources: [
+            // eslint-disable-next-line no-template-curly-in-string
             cdk.Fn.sub('arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/*-${apiId}-${envName}', {
-              apiId: ctx.api.apiId,
-              envName: ctx.synthParameters.amplifyEnvironmentName,
+              apiId: context.api.apiId,
+              envName: context.synthParameters.amplifyEnvironmentName,
             }),
           ],
         }),
-      );
-    }
-
-    this.generateResolvers(ctx);
-  }
-
-  protected createCustomProviderResource(scope: Construct, context: TransformerContextProvider): void {
-    // Policy that grants access to Create/Update/Delete DynamoDB tables
-    this.ddbManagerPolicy = new aws_iam.Policy(scope, 'CreateUpdateDeleteTablesPolicy');
-
-    const lambdaCode = aws_lambda.Code.fromAsset(
-      path.join(__dirname, '..', '..', '..', 'lib', 'resources', 'amplify-dynamodb-table', 'amplify-table-manager-lambda'),
-    );
-
-    // lambda that will handle DDB CFN events
-    const gsiOnEventHandler = new aws_lambda.Function(scope, ResourceConstants.RESOURCES.TableManagerOnEventHandlerLogicalID, {
-      runtime: aws_lambda.Runtime.NODEJS_18_X,
-      code: lambdaCode,
-      handler: 'amplify-table-manager-handler.onEvent',
-      timeout: Duration.minutes(14),
+      ],
     });
 
-    // lambda that will poll for provisioning to complete
-    const gsiIsCompleteHandler = new aws_lambda.Function(scope, ResourceConstants.RESOURCES.TableManagerIsCompleteHandlerLogicalID, {
-      runtime: aws_lambda.Runtime.NODEJS_18_X,
-      code: lambdaCode,
-      handler: 'amplify-table-manager-handler.isComplete',
-      timeout: Duration.minutes(14),
+    // Note: The isCompleteRole and onEventRole are similar enough that you might ask "why not just use a single role?"
+    // 1. Doing so creates a circular dependency between someCombinedRole <-> waiterStateMachine
+    // 2. The isCompleteHandler doesn't need permissions to invoke the waiterStateMachine.
+
+    // Role assumed by the isCompleteHandler.
+    // We want to avoid the auto-generated default policy for this to avoid unnecessary deployment time
+    // slowdowns, hence the `withPolicyUpdates()`
```

Comment:
Followup in future PR

```suggestion
    // slowdowns, hence the `withoutPolicyUpdates()`
```